src/analyzer/analyzermanager.cpp

```diff
@@ -0,0 +1,299 @@
+    }
+    if (shutdown) {
+        m_prioTrackQueue.clear();
+        QListIterator<AnalyzerWorker*> it2(m_foregroundWorkers);
+        while (it2.hasNext()) {
+            AnalyzerWorker* worker = it2.next();
+            worker->endProcess();
+            m_endingWorkers.append(worker);
+        }
+        //TODO: ensure that they are all forcibly stopped.
+    }
+}
+// Analyze it with a foreground worker. (foreground as in interactive, i.e. not a batch worker).
+void AnalyzerManager::analyseTrackNow(TrackPointer tio) {
+    if (m_batchTrackQueue.contains(tio)) {
+        m_batchTrackQueue.removeAll(tio);
+    }
+    //TODO: There's one scenario that we still miss: load on a deck a track that is currently
+    //being analyzed by the background worker. We cannot reuse the background worker, but we should discard its work.
+    if (!m_prioTrackQueue.contains(tio)) {
+        m_prioTrackQueue.append(tio);
+        if (m_foregroundWorkers.size() < m_MaxThreads) {
+            createNewWorker(false);
+            if (m_foregroundWorkers.size() + m_backgroundWorkers.size() > m_MaxThreads) {
+                AnalyzerWorker * backwork = m_backgroundWorkers.first();
+                backwork->pause();
+                //Ideally i would have done this on the slotPaused slot, but then i cannot 
+                //ensure i won't call pause twice for the same worker.
+                m_pausedWorkers.append(backwork);
+                m_backgroundWorkers.removeAll(backwork);
+            }
+        }
+    }
+}
+// This is called from the GUI for batch analysis.
+void AnalyzerManager::queueAnalyseTrack(TrackPointer tio) {
+    if (!m_batchTrackQueue.contains(tio)) {
+        m_batchTrackQueue.append(tio);
+        if (m_pausedWorkers.size() + m_backgroundWorkers.size() < m_MaxThreads) {
+            createNewWorker(true);
+        }
+    }
+}
+
+// This slot is called from the decks and samplers when the track is loaded.
+void AnalyzerManager::slotAnalyseTrack(TrackPointer tio) {
+    analyseTrackNow(tio);
+}
+
+
+//slot
+void AnalyzerManager::slotUpdateProgress(int workerIdx, struct AnalyzerWorker::progress_info* progressInfo) {
+    //Updates to wave overview and player status text comes from a signal emited from the track by calling setAnalyzerProgress.
+    progressInfo->current_track->setAnalyzerProgress(progressInfo->track_progress);
+    //These update the Analysis feature and analysis view.
+    emit(trackProgress(workerIdx, progressInfo->track_progress / 10));
+    if (progressInfo->track_progress == 1000) {
+        //Right now no one is listening to trackDone, but it's here just in case.
+        emit(trackDone(progressInfo->current_track));
+        //Report that a track analysis has finished, and how many are still remaining.
+        emit(trackFinished(m_backgroundWorkers.size() + m_batchTrackQueue.size() - 1));
+    }
+    //TODO: Which is the consequence of not calling clear?
+#if QT_VERSION < QT_VERSION_CHECK(5, 0, 0)
```

Comment:
Please use reset(). This conditional compilation is no longer necessary since we introduced a dedicated class for TrackPointer that handles the differences between QSharedPointer and std::shared_pointer internally.